tests/ccl_test_correlation.c

```diff
@@ -0,0 +1,391 @@
+    else
+      npoints++;
+
+    /*First time the tolerance is set. The tolerance is equal to the 
+     *expected error bar times CORR_ERR_FRACTION=0.5 (default) */
+    tol=gsl_spline_eval(spl_sigwt_dd_11,theta_in[ii],NULL);
+    if(fabs(wt_dd_11_h[ii]-wt_dd_11[ii])>tol*CORR_ERROR_FRACTION)
+      fraction_failed++;
+#ifdef _DEBUG
+    fprintf(output,"%.10e %.10e %.10e %.10e",theta_in[ii],wt_dd_11_h[ii],wt_dd_11[ii],tol);
+#endif //_DEBUG
+    //    tol=gsl_spline_eval(spl_sigwt_dd_12,theta_in[ii],NULL);
+    //    if(fabs(wt_dd_12_h[ii]-wt_dd_12[ii])>tol*CORR_ERROR_FRACTION)
+    //      fraction_failed++;
+#ifdef _DEBUG
+    //    fprintf(output," %.10e %.10e",wt_dd_12_h[ii],wt_dd_12[ii]);
+#endif //_DEBUG
+    tol=gsl_spline_eval(spl_sigwt_dd_22,theta_in[ii],NULL);
+    if(fabs(wt_dd_22_h[ii]-wt_dd_22[ii])>tol*CORR_ERROR_FRACTION)
+      fraction_failed++;
+#ifdef _DEBUG
+    fprintf(output," %.10e %.10e %.10e",wt_dd_22_h[ii],wt_dd_22[ii],tol);
+#endif //_DEBUG
+
+    tol=gsl_spline_eval(spl_sigwt_ll_11_pp,theta_in[ii],NULL);
+    if(fabs(wt_ll_11_h_pp[ii]-wt_ll_11_pp[ii])>tol*CORR_ERROR_FRACTION)
+      fraction_failed++;
+#ifdef _DEBUG
+    fprintf(output," %.10e %.10e %.10e",wt_ll_11_h_pp[ii],wt_ll_11_pp[ii],tol);
+#endif //_DEBUG
+    //    tol=gsl_spline_eval(spl_sigwt_ll_12_pp,theta_in[ii],NULL);
+    //    if(fabs(wt_ll_12_h_pp[ii]-wt_ll_12_pp[ii])>tol*CORR_ERROR_FRACTION)
+    //      fraction_failed++;
+#ifdef _DEBUG
+    //    fprintf(output," %.10e %.10e",wt_ll_12_h_pp[ii],wt_ll_12_pp[ii]);
+#endif //_DEBUG
+    tol=gsl_spline_eval(spl_sigwt_ll_22_pp,theta_in[ii],NULL);
+    if(fabs(wt_ll_22_h_pp[ii]-wt_ll_22_pp[ii])>tol*CORR_ERROR_FRACTION)
+      fraction_failed++;
+#ifdef _DEBUG
+    fprintf(output," %.10e %.10e %.10e",wt_ll_22_h_pp[ii],wt_ll_22_pp[ii],tol);
+#endif //_DEBUG
+
+    tol=gsl_spline_eval(spl_sigwt_ll_11_mm,theta_in[ii],NULL);
+    if(fabs(wt_ll_11_h_mm[ii]-wt_ll_11_mm[ii])>tol*CORR_ERROR_FRACTION)
+      fraction_failed++;
+#ifdef _DEBUG
+    fprintf(output," %.10e %.10e %.10e",wt_ll_11_h_mm[ii],wt_ll_11_mm[ii],tol);
+#endif //_DEBUG
+    //    tol=gsl_spline_eval(spl_sigwt_ll_12_mm,theta_in[ii],NULL);
+    //    if(fabs(wt_ll_12_h_mm[ii]-wt_ll_12_mm[ii])>tol*CORR_ERROR_FRACTION)
+    //      fraction_failed++;
+#ifdef _DEBUG
+    //    fprintf(output," %.10e %.10e",wt_ll_12_h_mm[ii],wt_ll_12_mm[ii]);
+#endif //_DEBUG
+    tol=gsl_spline_eval(spl_sigwt_ll_22_mm,theta_in[ii],NULL);
+    if(fabs(wt_ll_22_h_mm[ii]-wt_ll_22_mm[ii])>tol*CORR_ERROR_FRACTION)
+      fraction_failed++;
+#ifdef _DEBUG
+    fprintf(output," %.10e %.10e %.10e",wt_ll_22_h_mm[ii],wt_ll_22_mm[ii],tol);
+    fprintf(output,"\n");
+#endif //_DEBUG
+  }
+#ifdef _DEBUG
+  fclose(output);
+#endif //_DEBUG
+  
+  //Determine the fraction of points that failed the test
+  fraction_failed/=6*npoints;
+  printf("%lf %% ",fraction_failed*100);
+  //Check is this fraction is larger than we allow
+  ASSERT_TRUE((fraction_failed<CORR_FRACTION_PASS));

```

Comment:
All the comments I wrote in ccl_test_cls.c apply here too:
* where do the benchmarks come from?
* include nc-wl cross correlations.
* don't just allow some fraction to fail.